src/App.tsx

```diff
@@ -1,7 +1,37 @@
 import React from 'react';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <p className="App__message">The last pressed key is [Enter]</p>
-  </div>
-);
+interface State {
+  key: string | null;
+}
+
+export class App extends React.Component<{}, State> {
+  state: Readonly<State> = {
+    key: null,
+  };
+
+  componentDidMount() {
+    document.addEventListener('keyup', (event: KeyboardEvent) => {
+      this.setState({ key: event.key });
+    });
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('keyup', (event: KeyboardEvent) => {
+      this.setState({ key: event.key });
+    });
+  }
```

Comment:
it doesnt work :)

function, that you pass to addEventListener and to removeEventListener are two different functions.

that is why your removeEventListener will not work

you should to put it to variable and pass variable to both methods

Reply:
fixed, thank you!